### DISCUSSION TITLE
Project Family Defaults / Extending a Config

### DISCUSSION THREAD

**Discussion by michaeltlombardi** [estruyf/vscode-front-matter#392] [Q&A] [answered]
👋 I'm working on a hugo theme for indie analog game developers to help them put out great digital editions of their games - I was so glad to stumble on Front Matter this week because it already does probably about 80% of what I thought I was going to eventually have to implement in my own VS Code extension, which seemed like enormous work.

I've been reading through the docs and starting to investigate the tooling more seriously, but I'm wondering if there's already an established pattern for extending a basic configuration. My current thought is that I can provide a `frontmatter.json` file for my users and they can extend/override it in their project's `.vscode/settings.json` if I understand correctly? Is there a more sensible way, I'm not smart enough to see, or plans for an explicit extensibility model where a user can say their `frontmatter.json` extends another one?

With the method I'm noodling on, I think I'd also have to provide some scripts to help them keep their base configuration up to date / see what changes were made and available, which is fine - but if there's fewer steps or a smoother UX, I'm deeply interested in it!

**Comment by estruyf** [comment, marked as answer]
Let us continue with the issue you've created: https://github.com/estruyf/vscode-front-matter/issues/407
